### DISCUSSION TITLE
Can the Fragment tag be enabled to support data-testid ?

### DISCUSSION THREAD

**Discussion by wangcongyi** [testing-library/jest-dom#569] [Q&A] [answered]
![image](https://github.com/testing-library/jest-dom/assets/13843979/634499d0-035c-4e61-a689-4a8fe5d45c8c)
I know the `Fragment` render an empty tag, But I don't want to add meaningless `div` tags just for writing Jest tests.   
Wouldn't it be better if the Fragment could render actual DOM elements during run Jest tests?

**Comment by gnapse** [comment, marked as answer]
The `Fragment` React element does not render anything to the DOM (not even an empty tag, whatever that means).

jest-dom is not React-specific. It is not even React-aware. It is only DOM-aware. If something in React does not add anything to the DOM, it cannot have `data-` attributes.
